sefaria/helper/topic.py

```diff
@@ -107,6 +107,56 @@ def group_links_by_type(link_class, links, annotate_links, group_related):
                 grouped_links[link_type_slug]['pluralTitle'] = link_type.get('pluralDisplayName', is_inverse)
     return grouped_links
 
+def merge_props_for_similar_refs(curr_link, new_link):
+    # when grouping similar refs, make sure the source to display has the maximum curatedPrimacy of all the similar refs,
+    # as well as datasource and descriptions of all the similar refs
+    data_source = new_link.get('dataSource', None)
+    if data_source:
+        update_data_source_in_link(curr_link, new_link, data_source)
+        update_learning_team_data(curr_link, new_link, data_source)
+
+    if not curr_link['is_sheet']:
+        update_descriptions_in_link(curr_link, new_link)
+        update_curated_primacy(curr_link, new_link)
+    return curr_link
+
+def update_data_source_in_link(curr_link, new_link, data_source):
+    data_source_obj = library.get_topic_data_source(data_source)
+    curr_link['dataSources'][data_source] = data_source_obj.displayName
+    del new_link['dataSource']
+
+def update_learning_team_data(curr_link, new_link, data_source):
+    # in case the new_link was created by the learning team, we want to use ref of learning team link
+    # in the case when both links are from the learning team, use whichever ref covers a smaller range
+    if data_source != 'learning-team':
```

Comment:
we seem to be making a check in code for permissions. I understand that this is happening globally which isn't a great way to verify permissions. Lets rename the functions based on what the function does as opposed to who is doing the operation, and then have a decorator or some other check that verifies the person doing the operation is allowed to. The issue with "update_learning_team_data" is that I don't know what that function does, I only know who is supposed to be doing it